tests/upgrades/test_client.py

```diff
@@ -27,373 +28,303 @@
 from upgrade_tests.helpers.scenarios import dockerize
 from upgrade_tests.helpers.scenarios import get_entity_data
 
-from robottelo.api.utils import attach_custom_product_subscription
-from robottelo.api.utils import call_entity_method_with_timeout
-from robottelo.config import settings
-from robottelo.constants import DEFAULT_ORG
-from robottelo.test import APITestCase
+from robottelo.api.utils import enable_rhrepo_and_fetchid
+from robottelo.constants import DEFAULT_ARCHITECTURE
+from robottelo.constants import DEFAULT_CV
+from robottelo.constants import DEFAULT_SUBSCRIPTION_NAME
+from robottelo.constants import ENVIRONMENT
+from robottelo.constants import FAKE_0_CUSTOM_PACKAGE
+from robottelo.constants import FAKE_4_CUSTOM_PACKAGE
+from robottelo.constants import PRDS
+from robottelo.constants import REPOS
+from robottelo.constants import REPOSET
+from robottelo.constants.repos import FAKE_0_YUM_REPO
+from robottelo.constants.repos import FAKE_9_YUM_REPO
+from robottelo.test import settings
+
+
+# host machine for containers
+docker_vm = settings.upgrade.docker_vm
+# script in container /tmp requires SATHOST
+SATHOST = settings.server.hostname
+
+
+@pytest.fixture(scope='module')
+def module_lce_library(default_org):
+    """ Returns the Library lifecycle environment from chosen organization """
+    return (
+        entities.LifecycleEnvironment()
+        .search(query={'search': f'name={ENVIRONMENT} and organization_id={default_org.id}'})[0]
+        .read()
+    )
 
 
-def create_activation_key_for_client_registration(ak_name, client_os, org, environment, sat_state):
-    """Creates Activation key for client registration
+@pytest.fixture(scope='module')
+def rhel7_sat6tools_repo(default_org):
```

Comment:
```suggestion
def sat6tools_repo(default_org):
```

Lets not hardcode rhel7, this might change !